src/Workspaces/Core/Portable/Workspace/Host/PersistentStorage/AbstractPersistentStorage.cs

```diff
@@ -11,77 +11,34 @@ namespace Microsoft.CodeAnalysis.Host
 {
     internal abstract class AbstractPersistentStorage : IPersistentStorage
     {
-        private readonly IOptionService _optionService;
-        private readonly Action<AbstractPersistentStorage> _disposer;
-
-        private int _refCounter;
```

Comment:
Why is this no longer refcounted?

Reply:
Ref counting is being done by the service directly via a wrapper pattern rather than requiring each implementation to do it.

This code was pretty tightly coupled in that the storage object was refcounting directly. The single instance was disposed from potentially multiple consumers (ick), and that _disposer called into the service, which called back into the object to actually manipulate ref counts, The types were pretty impressively coupled as a result.

Now, Dispose() means this actually goes away. For multiple consumers, we hand out a wrapper type which forwards to the single instance, and each Dispose is tracked nicely with @sharwell's type.

In other words: this type now has sane lifetime semantics in the absence of a service managing it. I'm not sure what it was before.